### ISSUE TITLE
[v2 notifications] Query subscription with subscription/all API got 500 error

### ISSUE THREAD

**Issue by cherrycl** [edgexfoundry/edgex-go#3284]
# 🐞 Bug Report

### Affected Services

 The issue is located in: support-notifications

### Is this a regression?

 Yes, the previous version in which this bug was not present was: ....

### Description and Minimal Reproduction
1. Create a subscription with the following body.
```
POST /subscription API
[
   {
      "subscription":{
         "name":"Subscription-1616393640638-2",
         "description":"Test data for subscription",
         "categories":[
            "test"
         ],
         "labels":[
            "simple"
         ],
         "receiver":"test",
         "resendLimit":0,
         "resendInterval":"24h",
         "channels":[
            {
               "EmailAddresses":[
                  "123@123.com"
               ],
               "Type":"EMAIL"
            }
         ]
      },
      "apiVersion":"v2"
   }
]
```
2. Query the subscription using /subscription/all

## 🔥 Exception or Error
<pre><code>

 
API response:
{
    "apiVersion": "v2",
    "requestId": "",
    "message": "subscription format parsing failed from the database",
    "statusCode": 500
}

docker logs:
level=ERROR ts=2021-03-22T08:20:49.109155611Z app=edgex-support-notifications source=subscription.go:115 X-Correlation-ID=6f6c172e-b5ac-45d3-9f51-377e304bdc38 msg=" ->  -> subscription format parsing failed from the database -> json: cannot unmarshal object into Go struct field Subscription.Channels of type models.Address"
</code></pre>

## 🌍  Your Environment
**Deployment Environment:**

**EdgeX Version:** master

**Anything else relevant?**
All get methods returned 500 status code.

**Comment by cherrycl** [comment]
Delete subscription by /subscription/name/{name} failed with the same error.  
`level=ERROR ts=2021-03-22T09:30:48.149336011Z app=edgex-support-notifications source=subscription.go:287 X-Correlation-ID=0afe267b-48c8-4515-889d-48e6999cd450 msg=" -> fail to delete the subscription with name Subscription-3 ->  ->  -> object *models.Subscription format parsing failed from the database -> json: cannot unmarshal object into Go struct field Subscription.Channels of type models.Address"`

**Comment by cherrycl** [comment]
This issue is fixed by [go-mod-core-contracts PR #555](https://github.com/edgexfoundry/go-mod-core-contracts/pull/555).
